src/v/datalake/translation/partition_translator.cc

```diff
@@ -284,13 +259,22 @@ ss::future<> partition_translator::run_one_translation_iteration(
         co_await std::move(translation_f).finally([&cancellation_timer] {
             cancellation_timer.cancel();
         });
-    } catch (...) {
-        ex = std::current_exception();
+        _inflight_translation_state->as.check();
+    } catch (const translator_out_of_memory_error&) {
         vlog(
           _logger.warn,
-          "Translation attempt failed: {}, discarding state to reset "
-          "translation",
-          ex);
+          "Translation attempt ran into OOM, result will be flushed "
```

Comment:
nit: just for peace of mind i wonder if it's worth calling this something other than OOM, like it hit the reservation limit or something. OOM sounds a little scary? (not sure 🤔)